### PR TITLE
removed compilation errors, added libcdf

### PR DESCRIPTION
**Pull request by lambday**
- added incomplete_beta from libcdf (wrapper in CStatistics)
- also, added unit-tests, asserted against values computed from octave

@karlnapf please have a look!

### CODE REVIEW

File: src/shogun/mathematics/Statistics.cpp [hunk 1]
```diff
@@ -754,17 +755,43 @@ float64_t CStatistics::inverse_gamma_cdf(float64_t p, float64_t a,
...
-//	SG_SERROR("NOT IMPLEMENTED");
-//	return 0;
-//}
+float64_t CStatistics::incomplete_beta(float64_t a, float64_t b, float64_t x)
+{
+	float64_t y=1-x;
+	float64_t result=0;
+	float64_t result2=0;
+	int error_code=0;
+	beta_inc(&a, &b, &x, &y, &result, &result2, &error_code);
```
Comment (lambday) [inline src/shogun/mathematics/Statistics.cpp]: @karlnapf there is a look-up table or sort for some predefined a, b and x in cdflib. We can use that. Also, maybe for  `x < 0.5`, we can use symmetry - will check if that is faster.

File: examples/undocumented/libshogun/evaluation_cross_validation_classification.cpp [hunk 2]
```diff
@@ -105,7 +105,7 @@ void test_cross_validation()
 			splitting, eval_crit);
 
 	cross->set_num_runs(10);
-	cross->set_conf_int_alpha(0.05);
+//	cross->set_conf_int_alpha(0.05);
```
Comment (karlnapf) [inline examples/undocumented/libshogun/evaluation_cross_validation_classification.cpp]: These guys can be deleted. They are useless anyways :)

File: examples/undocumented/libshogun/mathematics_confidence_intervals.cpp [hunk 3]
```diff
@@ -17,6 +17,7 @@ using namespace shogun;
 
 void test()
 {
```
Comment (karlnapf) [inline examples/undocumented/libshogun/mathematics_confidence_intervals.cpp]: the whole file should go :)

File: examples/undocumented/libshogun/modelselection_combined_kernel_sub_parameters.cpp [hunk 4]
```diff
@@ -162,7 +162,7 @@ void modelselection_combined_kernel()
 
 	/* larger number of runs to have tighter confidence intervals */
```
Comment (karlnapf) [inline examples/undocumented/libshogun/modelselection_combined_kernel_sub_parameters.cpp]: the comment should be changed to "to have less noise"

File: src/shogun/mathematics/Statistics.cpp [hunk 5]
```diff
@@ -754,17 +755,43 @@ float64_t CStatistics::inverse_gamma_cdf(float64_t p, float64_t a,
...
+	if (error_code!=0)
+	{
+		switch (error_code)
+		{
+			case 1:
+				SG_SERROR("Params a (%f) or b (%f) cannot be negative!\n", a, b);
```
Comment (karlnapf) [inline src/shogun/mathematics/Statistics.cpp]: I think these guys should be checked via REQUIRE before, but then the error codes can basically be ignored right?

Comment (karlnapf) [inline src/shogun/mathematics/Statistics.cpp]: btw we should always print the value of the given numbers as in the REQUIRE elsehwere

**Comment by karlnapf** [comment]
I think this is good. I will merge and continue to edit myself

**Comment by yorkerlin** [comment]
@karlnapf @lambday 
Does this PR can pass the unit test about `lnormal_cdf` ?

**Comment by karlnapf** [comment]
yes, the test was never touched and passes

**Pull request merged**